lib/internal/preferences/AbstractPreferenceStore.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2019 Atomist, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    ConfigurationAware,
+    HandlerContext,
+} from "@atomist/automation-client";
+import { PreferenceStore } from "@atomist/sdm";
+
+export interface Preference {
+    key: string;
+    value: string;
+    ttl: number;
```

Comment:
If this can be `undefined`, should it be optional?

Reply:
It can be undefined but is required in the custom type